Tests/SwiftDocCUtilitiesTests/ConvertActionTests.swift

```diff
@@ -1664,6 +1664,89 @@ class ConvertActionTests: XCTestCase {
         }
     }
     
+    func testConvertActionProducesDeterministicOutput() throws {
+        // Pretty printing the output JSON also enables sorting keys during encoding
+        // which is required for testing if the conversion output is deterministic.
+        let priorPrettyPrintValue = shouldPrettyPrintOutputJSON
+        shouldPrettyPrintOutputJSON = true
+        defer {
+            shouldPrettyPrintOutputJSON = priorPrettyPrintValue
+        }
+        
+        let testBundleURL = try XCTUnwrap(
+            Bundle.module.url(
+                forResource: "TestBundle",
+                withExtension: "docc",
+                subdirectory: "Test Bundles"
+            )
+        )
+        let bundle = try Folder.createFromDisk(url: testBundleURL)
+        
+        func performConvertAction(outputURL: URL, testFileSystem: TestFileSystem) throws {
+            var action = try ConvertAction(
+                documentationBundleURL: bundle.absoluteURL,
+                outOfProcessResolver: nil,
+                analyze: false,
+                targetDirectory: outputURL,
+                htmlTemplateDirectory: Folder.emptyHTMLTemplateDirectory.absoluteURL,
+                emitDigest: false,
+                currentPlatforms: nil,
+                dataProvider: testFileSystem,
+                fileManager: testFileSystem
+            )
+            
+            _ = try action.perform(logHandle: .none)
+        }
+        
+        // We'll perform 3 sets of conversions to confirm the output is deterministic
+        for _ in 1...3 {
+            let testFileSystem = try TestFileSystem(
+                folders: [bundle, Folder.emptyHTMLTemplateDirectory]
+            )
+            
+            // Convert the same bundle three times and place the output in
+            // separate directories.
+            
+            try performConvertAction(
+                outputURL: URL(fileURLWithPath: "/1", isDirectory: true),
+                testFileSystem: testFileSystem
+            )
+            try performConvertAction(
+                outputURL: URL(fileURLWithPath: "/2", isDirectory: true),
+                testFileSystem: testFileSystem
+            )
+            
+            // Extract and sort the RenderJSON output of each conversion
+            
+            let firstConversionFiles = testFileSystem.files.lazy.filter { key, _ in
+                key.hasPrefix("/1/data/")
+            }.map { (key, value) in
+                return (String(key.dropFirst(2)), value)
```

Comment:
Nit, so that it's a bit clearer what we're dropping

```suggestion
                return (String(key.dropFirst("/1".count)), value)
```

Reply:
Good idea! Done.